nixos/modules/programs/fish.nix

```diff
@@ -84,9 +84,11 @@ in
 
       set fish_function_path $fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions
 
-      fenv source ${config.system.build.setEnvironment} > /dev/null ^&1
-      fenv source /etc/fish/foreign-env/shellInit > /dev/null
+      if status --is-login
+        fenv source ${config.system.build.setEnvironment} > /dev/null ^&1
```

Comment:
This could probably break commands via non-interactive ssh:

```
ssh foobar <command>
```

In `/etc/profile` we do instead:

```
if [ -n "$__ETC_PROFILE_SOURCED" ]; then return; fi                                                                                                                          
__ETC_PROFILE_SOURCED=1 
```

Something similar is probably preferred in fish.